engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -429,6 +433,12 @@ public void allocate(final String vmInstanceName, final VirtualMachineTemplate t
 
         final VirtualMachineProfileImpl vmProfile = new VirtualMachineProfileImpl(vmFinal, template, serviceOffering, null, null);
 
+        Long rootDiskSize = rootDiskOfferingInfo.getSize();
+        if (vm.getType().isUsedBySystem() && SystemVmRootDiskSize.value() != null) {
+            rootDiskSize = SystemVmRootDiskSize.value();
```

Comment:
@ustcweizhou should there be a check for `> 0` ?

Reply:
@rhtyd added check.